rust/src/live.rs

```diff
@@ -468,34 +469,30 @@ pub(crate) fn transaction_apply_live(
 
     // Gather the current diff of /etc - we need to avoid changing
     // any files which are locally modified.
-    let task = crate::ffi::progress_begin_task("Computing /etc diff to preserve");
-    let config_diff = {
+    let config_diff = progress_task("Computing /etc diff to preserve", || -> Result<_> {
         let usretc = &rootfs_dfd.sub_dir("usr/etc")?;
         let etc = &rootfs_dfd.sub_dir("etc")?;
-        crate::dirdiff::diff(usretc, etc)?
-    };
+        crate::dirdiff::diff(usretc, etc)
+    })?;
     println!("Computed /etc diff: {}", &config_diff);
-    std::mem::drop(task);
 
     // The heart of things: updating the overlayfs on /usr
-    let task = crate::ffi::progress_begin_task("Updating /usr");
-    apply_diff(repo, &diff, &target_commit, &openat::Dir::open("/usr")?)?;
-    std::mem::drop(task);
+    progress_task("Updating /usr", || -> Result<_> {
+        apply_diff(repo, &diff, &target_commit, &openat::Dir::open("/usr")?)
+    })?;
 
     // The other important bits are /etc and /var
-    let task = crate::ffi::progress_begin_task("Updating /etc");
-    update_etc(
-        repo,
-        &diff,
-        &config_diff,
-        &sepolicy,
-        &target_commit,
-        &openat::Dir::open("/etc")?,
-    )?;
-    std::mem::drop(task);
-    let task = crate::ffi::progress_begin_task("Running systemd-tmpfiles for /run and /var");
-    rerun_tmpfiles()?;
```

Comment:
Looks like we lost this.

Reply:
It'ss still there, it's just a one-liner now: https://github.com/coreos/rpm-ostree/pull/2667/files#diff-c4c5d857a0d0516334ffb1d582be77876848adc28f3464d7dc58b5e3464f9c67R495

Reply:
(Also, CI covers this)